api/v1beta1/hostedcluster_types.go

```diff
@@ -864,6 +864,15 @@ type AWSPlatformSpec struct {
 	// +kubebuilder:default=Public
 	// +optional
 	EndpointAccess AWSEndpointAccessType `json:"endpointAccess,omitempty"`
+
+	// AdditionalAllowedPrincipals specifies a list of additional allowed principal ARNs
+	// to be added to the hosted control plane's VPC Endpoint Service to enable additional
+	// VPC Endpoint connection requests to be automatically accepted.
+	// See https://docs.aws.amazon.com/vpc/latest/privatelink/configure-endpoint-service.html
+	// for more details around VPC Endpoint Service allowed principals.
+	//
+	// +optional
```

Comment:
I think this could name this just `allowedPrincipals`? wdyt?

Reply:
I was thinking `additional` only because the control plane operator's IAM role is always an allowed principal, but this field would allow users to specify others